### PR TITLE
Upload theme

### PR DESCRIPTION
**Pull request by jouni**
This change is [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-material-theme/47)

### CODE REVIEW

File: vaadin-upload.html [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+        opacity: 0;
+        transform: translate(-50%, -50%) scale(0);
+        transition: transform 0s cubic-bezier(0.05, 0.8, 0.5, 1), opacity 0.4s linear;
+        transition-delay: 0.4s, 0s;
+      }
+
+      :host([dragover-valid])::before {
+        transform: translate(-50%, -50%) scale(6);
+        opacity: 0.1;
+        transition-duration: 2s, 0.1s;
```
Comment (platosha) [inline vaadin-upload.html]: This is slo-o-ow and does not cover the entire host if it’s wide.

Take a look how quick they are in the guide: https://material.io/design/motion/speed.html#controlling-speed

File: vaadin-upload.html [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+        line-height: 1;
+      }
+
+      [part="start-button"],
+      [part="retry-button"],
+      [part="clear-button"] {
+        flex: none;
+        color: var(--material-secondary-text-color);
+        line-height: 1;
+        padding: 4px;
```
Comment (platosha) [inline vaadin-upload.html]: Quite small touch target size. Increasing padding up to 8px on each side does not seem to harm much here.

Not a blocker though, we can adjust this later.

File: vaadin-upload.html [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+      [part="clear-button"]:hover {
+        color: inherit;
+      }
+
+      [part="done-icon"]::before,
+      [part="warning-icon"]::before,
+      [part="start-button"]::before,
+      [part="retry-button"]::before,
+      [part="clear-button"]::before {
+        vertical-align: -0.05em;
```
Comment (platosha) [inline vaadin-upload.html]: Well, if all the icons baseline should be adjusted, should we move the baseline in the icon font instead? This is pretty easy to do now that we use our script to create it.

File: vaadin-upload.html [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+
+<dom-module id="material-upload-file" theme-for="vaadin-upload-file">
+  <template>
+    <style>
+      :host {
+        padding: 8px 0;
+      }
+
+      :host(:not(:first-child)) {
+        border-top: 1px solid var(--material-divider-color);
```
Comment (platosha) [inline vaadin-upload.html]: Looks like a grid of files with these lines.

Two-line lists in the guide, typically, don’t use horizontal borders between items: https://material.io/design/components/lists.html#types

Let's consider removing them maybe?

**Comment by platosha** [comment]
Well, it’s improvement over what we have, so good to merge.

I really really miss smooth transitions on file done and remove though. They are actually helpful to give some time to realise what happens to the file. Now, because the transition is not smooth, a small change sometimes feels like a completely different list to read.

File: vaadin-upload.html (hunk 4 above)
Comment (jouni) [reply vaadin-upload.html]: Good point. Removed the dividers.

File: vaadin-upload.html (hunk 3 above)
Comment (jouni) [reply vaadin-upload.html]: Postponed.

File: vaadin-upload.html (hunk 2 above)
Comment (jouni) [reply vaadin-upload.html]: Increased the size (with 8px padding).

File: vaadin-upload.html (hunk 1 above)
Comment (jouni) [reply vaadin-upload.html]: Increased the size and animation speed. Probably not excellent yet, but should feel a bit better.

**Pull request merged**